### PR TITLE
Pal 999 keep common library

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-@Library('jenkinsfile-lib')_
+@Library('jenkinsfile-lib@feature/PAL-999-keep-common-library')_
```
Comment (developer01189998819991197253) [inline Jenkinsfile:16]: revert jenkinsfile before merge

**Review by developer01189998819991197253 — COMMENTED** [review]
revert licenses but otherwise approved

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopDataReader.java [hunk 2]
```diff
@@ -48,13 +47,17 @@
 
     private FileSystem fs;
 
+    public HadoopDataReader(final Configuration configuration) throws IOException {
```
Comment (pd104923) [inline hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopDataReader.java:50]: Does this need some JavaDoc?

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import uk.gov.gchq.palisade.service.resource.service.ResourceService;
+
+import java.io.IOException;
+
+@Configuration
+public class HadoopConfiguration {
```
Comment (pd104923) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopConfiguration.java:28]: This class will need JavaDoc

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+package uk.gov.gchq.palisade.service.data.hadoop;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+
+import uk.gov.gchq.palisade.service.data.reader.DataReader;
+
+import java.io.IOException;
+
+public class HadoopConfiguration {
```
Comment (pd104923) [inline hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopConfiguration.java:26]: This class will need some JavaDoc

**Review by pd104923 — CHANGES_REQUESTED** [review]

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopDataReader.java (hunk 2 above)
Comment (developer01189998819991197253) [reply hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopDataReader.java:50]: added

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopConfiguration.java (hunk 3 above)
Comment (developer01189998819991197253) [reply hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopConfiguration.java:28]: added

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopConfiguration.java (hunk 4 above)
Comment (developer01189998819991197253) [reply hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopConfiguration.java:26]: added

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import uk.gov.gchq.palisade.service.resource.service.ResourceService;
+
+import java.io.IOException;
+
+/**
+ * A Spring Hadoop Configuration class, creating the necessary beans for a Hadoop data reader
```
Comment (pd104923) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopConfiguration.java:28]:
```suggestion
 * A Spring Hadoop Configuration class, creating the necessary beans for a Hadoop resource service
```

**Review by pd104923 — CHANGES_REQUESTED** [review]

**Review by pd104923 — APPROVED** [review]

**Comment by ac74475** [comment]
can be merged once corresponding palisade services PR is merged

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+    /**
+     * Bean implementation for {@link HadoopDataReader} which extends {@link uk.gov.gchq.palisade.service.data.reader.SerialisedDataReader} and is used for setting hadoopConfigurations and reading raw data.
+     *
+     * @param configuration a hadoop configuration specifying the target cluster
+     * @return a new instance of {@link HadoopDataReader}
+     * @throws IOException ioException
+     */
+    @Bean
+    @ConditionalOnProperty(prefix = "data", name = "implementation", havingValue = "hadoop", matchIfMissing = true)
```
Comment (dev930018) [inline hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopConfiguration.java:44]:
```suggestion
    @ConditionalOnProperty(prefix = "data", name = "implementation", havingValue = "hadoop")
```
Simple/default/null should matchifmissing, not this

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopResourceDetails.java [hunk 7]
```diff
@@ -1,190 +1,190 @@
...
-    @Generated
-    public String toString() {
-        return new StringJoiner(", ", HadoopResourceDetails.class.getSimpleName() + "[", "]")
-                .add("fileName='" + fileName + "'")
-                .add("type='" + type + "'")
-                .add("format='" + format + "'")
-                .toString();
-    }
-}
+/*
```
Comment (dev930018) [inline hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopResourceDetails.java:1]: Check line-endings are reasonable considering this diff

File: hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopConfiguration.java (hunk 6 above)
Comment (pd104923) [reply hadoop-reader/src/main/java/uk/gov/gchq/palisade/service/data/hadoop/HadoopConfiguration.java:44]: Removed the "matchIfMissing"

File: hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopResourceDetails.java (hunk 7 above)
Comment (pd104923) [reply hadoop-resource/src/main/java/uk/gov/gchq/palisade/service/resource/hadoop/HadoopResourceDetails.java:1]: The line endings look good.

File: NOTICES.md [hunk 8]
```diff
@@ -1,29 +1,19 @@
...
+* Gson ([com.google.code.gson:gson:2.8.6](https://github.com/google/gson/gson))
 * Apache Hadoop Common ([org.apache.hadoop:hadoop-common:3.2.1](no url defined))
 * AssertJ fluent assertions ([org.assertj:assertj-core:3.18.0](https://assertj.github.io/doc/assertj-core/))
-* common ([uk.gov.gchq.palisade:common:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-common))
-* readers-common ([uk.gov.gchq.palisade:readers-common:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-readers/tree/develop/readers-common))
-* shade-hadoop-client ([uk.gov.gchq.palisade:shade-hadoop-client:0.5.0-SNAPSHOT](https://github.com/gchq/Palisade-readers/tree/develop/shade-hadoop-client))
-
-### [BSD License 3](./licenses/bsd_license_3.html):
-* Hamcrest ([org.hamcrest:hamcrest:2.1](http://hamcrest.org/JavaHamcrest/))
+* data-service ([uk.gov.gchq.palisade:data-service:0.5.0-SNAPSHOT](no url defined))
```
Comment (developer01189998819991197253) [inline NOTICES.md:8]: add data service url to the notices, like done for the resource service

File: licenses/bsd_license_3.html [hunk 9]
```diff
@@ -1,365 +0,0 @@
-<!DOCTYPE html>
```
Comment (developer01189998819991197253) [inline licenses/bsd_license_3.html:1]: revert licence changes

**Review by developer01189998819991197253 — CHANGES_REQUESTED** [review]

File: NOTICES.md (hunk 8 above)
Comment (pd104923) [reply NOTICES.md:8]: url added

**Review by developer01189998819991197253 — APPROVED** [review]

**Pull request merged**